Release 1.0.0

vmkmym opened the pull request:
## What is this PR? :mag:
- 메이트립 버그 해결
- 추가 기능 반영

## Changes :memo:
- #47 
- #48 
- #49 
- refactor : 동행글 상대방 프로필 조회 api 연동
- feat : 동행글 필터링 기능 추가
- feat : Refresh Token, Access Token 삭제 로직 추가
- feat : 백문백답 기능 추가
- design : 홈화면 알림 아이콘 색상 수정
- design : 자잘한 UI 미반영 사항 수정
- fix : FAB 버튼 안 보이는 문제 해결

## Screenshot :camera:

uselessnaming committed:
Design: 넓이 변경 (60 >> 80)

uselessnaming committed:
Feat: 앱 Loading Component 추가

uselessnaming committed:
Design: Loading 상태 ui 반영

uselessnaming committed:
Design: component 배경 색 변경 (inactive >> gray03)

uselessnaming committed:
Design: 테두리 제거

uselessnaming committed:
Design: 아이콘 크기 조절

uselessnaming committed:
Feat: 백문백답 service 추가

uselessnaming committed:
Feat: dto 추가

uselessnaming committed:
Rename: 클래스 명 변경

vmkmym committed:
fix: 홈화면 게시글 중복 로드 문제 해결

- PRIVACY_URL를 local.properties에 추가하고 localProperties 경로 수정
- 게시글 화면 이동 후 돌아올 때 게시글 목록 무한 복사 문제 해결

vmkmym committed:
fix: 이미지 첨부 시 앱이 종료 되는 문제 해결

- 업로드하려는 이미지 파일의 크기가 서버에서 허용하는 최대 크기보다 크기 때문에 발생하는 오류가 있어서 사진을 서버에 업로드할 때 크기를 줄여서 업로드하는 로직으로 수정하여 해결
- 여행일정 재선택 아이콘이 modifier 영역에 묻혀 보이지 않아 길이 수정

uselessnaming committed:
Feat: 백문백답 data store 추가

uselessnaming committed:
Feat: 백문백답 repository 추가

uselessnaming committed:
Feat: 백문백답 service 추가

uselessnaming committed:
Feat: DI 설정

uselessnaming committed:
Feat: DI 설정

uselessnaming committed:
Feat: 백문백답 view model 추가

uselessnaming committed:
Feat: 파라미터 타입 변경 (not null >> nullable)

uselessnaming committed:
Feat: view model 연결

uselessnaming committed:
Fix: 개수 반영 안되는 오류 수정

uselessnaming committed:
Merge branch 'release-1.0.0' of https://github.com/dnd-side-project/dnd-11th-1-android into release-1.0.0

vmkmym committed:
fix: 입력값 검증 로직 추가, 게시 버튼 활성화 로직

vmkmym committed:
Merge branch 'release-1.0.0' of https://github.com/dnd-side-project/dnd-11th-1-android into release-1.0.0

vmkmym committed:
fix: 입력값 검증 로직 추가, 게시 버튼 활성화 로직

vmkmym committed:
fix: 동행글 작성 화면 진입 시 이전 작성 내용이 유지되는 문제

- ViewModel을 Activity 스코프로 유지하면서, 사용자가 FAB 버튼을 눌러 PostBoardScreen으로 이동할 때 ViewModel의 상태를 초기화
- PostBoardViewModel에 모든 상태를 초기화하는 resetState() 함수
- NavController의 LifecycleListener를 활용하여 ViewModel 상태 초기화

uselessnaming committed:
Chore: local properties 설정

uselessnaming committed:
Fix: paging 오류 수정

uselessnaming committed:
Style: 코드 간소화

uselessnaming committed:
Design: 중복 ui 제거

uselessnaming commented:
release 브랜치에서 develop으로 다시 PR 인건가요?

uselessnaming commented:
release 브랜치로 계속 push하는 것 아니었나용?

uselessnaming committed:
Fix: api 태그 mismatch 수정 (get >> post)

uselessnaming committed:
Style: 공백 제거

uselessnaming committed:
Fix: api 태그 mismatch 수정 (get >> post)

uselessnaming committed:
Refactor: ui state 로직 변경

uselessnaming committed:
Fix: api 어노테이션 오류 수정 (get >> post)

uselessnaming committed:
Fix: api 어노테이션 오류 수정 (get >> post)

uselessnaming committed:
Fix: api 어노테이션 오류 수정 (get >> post)

vmkmym commented:
계속 개발하게 되지 않나요?
배포 후에 작업이 없다면 develop 브랜치에 반영하지 않아도 될 것 같아요
그 이후 작업이 있다면 QA 작업 후 반영된 커밋도 반영해야 해서 pull rq 만들었습니다
git flow가 이렇게 작업하는 걸로 알고 있는데 어떻게 하시고 싶으신가요??

uselessnaming commented:
아 저희 수정 작업이 끝났다고 생각이 되신건가요?
develop으로의 PR은 배포 준비가 완료된 후에 해도 괜찮을 것 같아요!!
아니면 다른 생각이 있으신가요?

uselessnaming committed:
Fix: 파라미터 수정 (body >> Query)

uselessnaming committed:
Chore: navigation 의존성 추가

uselessnaming committed:
Refactor: 성능 개선 (recomposition 감소)

uselessnaming committed:
Remove: 불필요 dto 정리

uselessnaming committed:
Feat: QnA 100 response dto 추가

uselessnaming committed:
Rename: response dto 이름 변경 (QnARequestItem >> QnAItemDto)

uselessnaming committed:
Feat: null 처리

uselessnaming committed:
Fix: get QnA api request 파라미터 및 response 형식 변경

uselessnaming committed:
Fix: 추가한 qna 삭제할 경우 발생하는 오류 삭제

uselessnaming committed:
Fix: 추가한 qna 삭제할 경우 발생하는 오류 수정

uselessnaming committed:
Rename: dto 이름 변경 (QnARequestItem >> QnAItemDto)

vmkmym commented:
오 그러면 나중에 한 번에 pr할 때는 pr 내용을 같이 적어야 하는건가요?
나중에 한 번에 해도 상관 없습니다~

uselessnaming commented:
아 그럼 완료되고 나서 PR 닫도록 하겠습니당

uselessnaming commented:
@vmkmym 그러면 현재 개발 완료된 부분들을 release에 push 하신건가요?

vmkmym committed:
fix: 동행신청 후 버튼 비활성화 로직 추가, boardId 전달 문제 해결

uselessnaming committed:
Fix: index out of range 오류 수정

vmkmym committed:
refactor: 유저 프로필 조회 API 수정

- userId 기반으로 유저의 프로필을 상세하게 조회 가능

vmkmym committed:
refactor: 유저 프로필 상세 화면 네비게이션 수정

- userId 기반으로 유저의 프로필을 상세하게 조회 가능

vmkmym committed:
refactor: 알림 화면 관련 프로필 조회 코드 수정

vmkmym committed:
refactor: 유저 프로필 조회 화면 비즈니스 로직 및 ui 수정

vmkmym committed:
chore: 빌드를 위한 경로 수정

- load(project.file("local.properties").inputStream()) 이렇게 하면 빌드가 안되고     load(project.rootProject.file("local.properties").inputStream()) 해야 빌드가 되어서 잠시 커밋

vmkmym committed:
Merge branch 'release-1.0.0' of https://github.com/dnd-side-project/dnd-11th-1-android into release-1.0.0

vmkmym committed:
design: 필터 이미지 및 아이콘 추가

vmkmym committed:
design: 동행라운지 필터 아이콘 추가

vmkmym committed:
feat: 동행 시작 여부에 따른 동행글 목록 조회 API 연동

uselessnaming committed:
Chore: local 파일 연결 수정

uselessnaming committed:
Fix: 중복 방지

uselessnaming committed:
Chore: 자동 back up 설정 false

uselessnaming committed:
Fix: DELETE -> POST 어노테이션 변경

uselessnaming committed:
Comment: token log 값 출력

uselessnaming committed:
Design: 아이콘 변경

uselessnaming committed:
Fix: 탈퇴 flow 순서 오류 수정

uselessnaming committed:
Design: top bar 제거 화면 추가

uselessnaming committed:
Fix: 탈퇴 flow 순서 오류 수정

uselessnaming committed:
Comment: auth token log 출력

uselessnaming committed:
Feat: logout 기능 추가

uselessnaming committed:
Feat: logout 기능 추가

uselessnaming committed:
Feat: 토큰 삭제 로직 추가

uselessnaming committed:
Style: 공백 제거

uselessnaming committed:
Design: 채소 아이콘 변경

uselessnaming committed:
Design: 아이콘 크기 변경

vmkmym committed:
Merge branch 'release-1.0.0' of https://github.com/dnd-side-project/dnd-11th-1-android into release-1.0.0

vmkmym committed:
refactor: 동행 시작 여부에 따른 동행글 목록 조회 API 업데이트

vmkmym committed:
design: 알림 아이콘 색상 수정

vmkmym committed:
feat: 동행글 목록 필터링 다이얼로그 추가

vmkmym committed:
feat: 동행라운지 필터링 로직 추가

vmkmym committed:
feat: 동행글 필터링 로직 업데이트

vmkmym committed:
feat: 동행글 필터링 관련 함수 추가

- updateFilter 함수는 필터링 된 동행글 목록을 불러옴
- updateSelectedOption 함수는 선택한 옵션을 업데이트 함
- FilterState 클래스는 필터 상태를 반영하는 데이터 클래스
- filteredBoardList 함수는 동행 시작 여부에 따른 동행글 목록 조회 함수

vmkmym committed:
feat: 동행글 리스트에 반영할 intent class 추가

vmkmym committed:
fix: 사라진 fab 버튼 다시 보이게 함

vmkmym commented:
@uselessnaming 수정 사항 본문에 추가해주시면 좋을 것 같습니다!

uselessnaming commented:
넴~

uselessnaming committed:
Merge branch 'release-1.0.0' of https://github.com/dnd-side-project/dnd-11th-1-android into release-1.0.0

uselessnaming committed:
Design: 색상 통일

vmkmym committed:
chore: localProperties 경로 수정

vmkmym committed:
Merge branch 'release-1.0.0' of https://github.com/dnd-side-project/dnd-11th-1-android into release-1.0.0

vmkmym committed:
fix: 동행신청 시 동행신청 버튼 비활성화가 앱을 껐다 켜면 적용되지 않는 버그 해결

- 동행신청 버튼의 상태가 앱을 재시작할 때 유지되지 않기 때문에 발생한 문제
- 동행신청 상태를 영구적으로 저장하고 불러오는 로직을 추가함
- ProfileViewModel에 SharedPreferences를 사용하여 동행신청 상태를 저장하고 불러오는 메서드를 추가

vmkmym committed:
refactor: 쓰지 않는 메서드 제거
